haas/model.py

```diff
@@ -371,6 +268,49 @@ def get_port_networks(self, ports):
         and have ``session`` just return ``self``.
         """
 
+class Obm(AnonModel):
+    """Obm superclass supporting various drivers 
+
+    related to out of band management of servers.
+    """
+    type = Column(String, nullable=False)
+
+    __mapper_args__ = {
+            'polymorphic_on': type
+            }
+    @staticmethod
+    def validate(kwargs):
+        """Verify that ``kwargs`` is a legal set of keywords args to ``__init__``
+
+        Raise a ``schema.SchemaError`` if the  ``kwargs`` is invalid. 
+        Note well: This is a *static* method; it will be invoked on the class.
+        """     
+        assert False, "Subclasses MUST override the validate method "
+
+    def power_cycle(self):
+        """Power cycles the node.
+
+        Exact implementation to left to the subclasses.
+        """     
+        assert False, "Subclasses MUST override the power_cycle method "
+
+    def start_console(self):
+        """Starts logging to the console. """
+        assert False, "Subclasses MUST override the start_console method" 
+
+    def stop_console(self):
+        """Stops console logging. """
+        assert False, "Subclasses MUST override the stop_console method"
+
+    def delete_console(self):
+        assert False, "Subclasses MUST override the delete_console method"
+
+    def get_console(self):
+        assert False, "Subclasses MUST override the get_console method"
+
+    def get_console_log_filename(self):
+        assert False, "Subclasses MUST override this method"
```

Comment:
Not according to the diff. I checked out your branch locally to be sure; this is unchanged.